### ISSUE TITLE
dataset

### ISSUE THREAD

**Issue by TryHard-LL** [ZhiyuanDang/DCDC#1]
Hi,
Can you share me with your imagenet-dogs datasets?
Thanks.

**Comment by ZhiyuanDang** [comment]
Hi @HBU-Lin-Li ,

I am afraid that I don't get the right to distribute the dataset. Maybe you need to apply for the Access Permission of ImageNet to use the data.

**Comment by TryHard-LL** [comment]
Hi @ZhiyuanDang ,

I have a question on ' The precondition of this statement is that we should keep one positive sample and N −1 negative samples.
However, for the smaller class number with large batch size cases, this precondition not holds any more.'
It means that if the smaller class number with large batch size, it may include more examples of the same kind?And then it may destroy the representations?
Now , I think that this is conflict with previous work....

Thanks.

**Comment by ZhiyuanDang** [comment]
Hi @HBU-Lin-Li ,

That does not conflict with previous work. Actually, more work finds that the existence of the false-positive sample in mini-batch, such as BYOL and more recent work. They only consider the positive and its augmentation samples, ignore the negative sample, and still achieve high performance.

Since the contrastive work only pre-train on ImageNet, some insights may not be suitable in the small datasets such as CIFAR-10/100.

**Comment by TryHard-LL** [comment]
@ZhiyuanDang, Thanks for your reply.

When we use SSL to solve clustering tasks, we utilize the contrastive loss.In extreme cases we only build positive example pairs by two differernt transformations of one image, this can't bring in false-positive sample in mini-batch.However, this can only bring in more hard-negative samples when the batch size is larger.If we use  smaller batch size, it may be bring in less hard-negative samples, and then may improve the performance.
Is that right?

**Comment by TryHard-LL** [comment]
@ZhiyuanDang, Thanks for your reply.

When we use SSL to solve clustering tasks, we utilize the contrastive loss.In extreme cases we only build positive example pairs by two differernt transformations of one image, this can't bring in false-positive sample in mini-batch.However, this can only bring in more hard-negative samples when the batch size is larger.If we use  smaller batch size, it may be bring in less hard-negative samples that will bring in less noise, and then may improve the performance.
Is that right?

**Comment by ZhiyuanDang** [comment]
The `false-positive sample` in the last reply should be` false-negative samples` (samples which in the same class of the positive sample).  That is a typo.

Hard-negative samples are acceptable for the clustering tasks, but the false-negative samples may degenerate the performance.

Larger batch size more bring more true and false-negative samples, especially in smaller class number.

As for larger class numbers such as TinyImageNet, it is fine to adopt a large batch size.

**Comment by TryHard-LL** [comment]
Thanks for your help.

**Comment by Wudiyinyin** [comment]
Hi, I have a question about the code. While I was running the code, I found that the parameters of register_module, register_class, and register_class under class _META_(type): in DCDC-main\lib\core didn't match the number of parameters that should be passed in in the method, which made the code unable to run.
I hope you can help me solve it. Thanks.
